cmd/goexpander/expander/expander_test.go

```diff
@@ -0,0 +1,275 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package expander
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/kubernetes/helm/pkg/chart"
+	"github.com/kubernetes/helm/pkg/common"
+	"github.com/kubernetes/helm/pkg/expansion"
+)
+
+// content provides an easy way to provide file content verbatim in tests.
+func content(lines []string) []byte {
+	return []byte(strings.Join(lines, "\n") + "\n")
+}
+
+func testExpansion(t *testing.T, req *expansion.ServiceRequest,
+	expResponse *expansion.ServiceResponse, expError string) {
+	backend := NewExpander()
+	response, err := backend.ExpandChart(req)
+	if err != nil {
+		message := err.Error()
+		if expResponse != nil || !strings.Contains(message, expError) {
+			t.Fatalf("unexpected error: %s\n", message)
+		}
+	} else {
+		if expResponse == nil {
+			t.Fatalf("expected error did not occur: %s\n", expError)
+		}
+		if !reflect.DeepEqual(response, expResponse) {
+			message := fmt.Sprintf(
+				"want:\n%s\nhave:\n%s\n", expResponse, response)
+			t.Fatalf("output mismatch:\n%s\n", message)
+		}
+	}
+}
+
+var goExpander = &chart.Expander{
+	Name:       "GoTemplating",
+	Entrypoint: "templates/main.py",
+}
+
+func TestEmpty(t *testing.T) {
+	testExpansion(
+		t,
+		&expansion.ServiceRequest{
+			ChartInvocation: &common.Resource{
+				Name: "test_invocation",
+				Type: "gs://kubernetes-charts-testing/Test-1.2.3.tgz",
+			},
+			Chart: &chart.Content{
+				Chartfile: &chart.Chartfile{
+					Name:     "Test",
+					Expander: goExpander,
+				},
+			},
+		},
+		&expansion.ServiceResponse{
+			Resources: []interface{}{},
+		},
+		"", // Error
+	)
+}
+
+func TestSingle(t *testing.T) {
+	if os.Getenv("CIRCLECI") == "true" {
+		// CircleCI seems to be using an old version of Go that has buggy templates.
```

Comment:
I haven't been able to install that version of go to try this out on yet.  It may be a bug, or just the old version had different behavior.  Do we need to support go 1.5.3?  Are any of us using that version?